src/index.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-env mocha */
+'use strict'
+
+module.exports = (common) => {
+  describe('interface-peer-discovery', () => {
+    let discovery
+
+    before(() => {
+      discovery = common.setup()
+    })
+
+    after(() => common.teardown && common.teardown())
+
+    it('can start the service', async () => {
+      await discovery.start()
+    })
+
+    it('can start and stop the service', async () => {
+      await discovery.start()
+      await discovery.stop()
+    })
+
+    it('should not fail to stop the service if it was not started', async () => {
+      await discovery.stop()
+    })
+
+    it('should not fail to start the service if it is already started', async () => {
+      await discovery.start()
+      await discovery.start()
+    })
```

Comment:
`discovery.on` and `discovery.removeListener` should get added as well, but those can wait until we add a proper test suite here, since they'll get tested as a part of that.

Reply:
Yap, I plan to add them in a next step, once we are working more closely with all the other peer discovery protocols.

Things are slightly different for each of them, and I need to have a better context on all of them to create a generic test setup for all. This way, I think we can start with this for the `*-star`, and then upgrade them once we get into `dht` and `mdns` ones